database/postgres/dal/jwt.py

```diff
@@ -0,0 +1,84 @@
+from sqlalchemy.orm import Session
+from database.models.token_blocklist import TokenBlocklist
+from database.postgres.dto.jwt import jwtDTO
+from sqlalchemy.exc import SQLAlchemyError
+from typing import Optional, List
+from datetime import datetime
+
+class jwtDAL:
+    def __init__(self, db_session: Session):
+        self.db_session = db_session
+
+    def save_jwt(self, jwt_dto: jwtDTO) -> Optional[int]:
+        try:
+            jwt_entry = self.get_jwt_by_id(jwt_dto.id) if jwt_dto.id else None
+            if jwt_entry:
+                jwt_entry.user_id = jwt_dto.user_id
+                jwt_entry.jti = jwt_dto.jti
+                jwt_entry.token_type = jwt_dto.token_type
+                jwt_entry.revoked = jwt_dto.revoked
+                jwt_entry.expires_at = jwt_dto.expires_at
+                jwt_entry.updated_at = datetime.utcnow()
+            else:
+                jwt_entry = TokenBlocklist(**jwt_dto.dict(exclude={"id", "updated_at"}), updated_at=datetime.utcnow())
+                self.db_session.add(jwt_entry)
+
+            self.db_session.commit()
+            self.db_session.refresh(jwt_entry)
+            return jwt_entry.id
+        except SQLAlchemyError as e:
+            self.db_session.rollback()
```

Comment:
We only have one operation happening here (which is a transaction on its own), so I don't see any reason for this rollback here.